Sources/NIOCore/EventLoopFuture.swift

```diff
@@ -178,30 +178,52 @@ public struct EventLoopPromise<Value> {
     ///
     /// - parameters:
     ///     - value: The successful result of the operation.
+    @preconcurrency
     @inlinable
-    public func succeed(_ value: Value) {
+    public func succeed(_ value: Value) where Value: Sendable {
         self._resolve(value: .success(value))
     }
 
+    /// Deliver a successful result to the associated `EventLoopFuture<Value>` object.
+    ///
+    /// - Note: The call to this method must happen on the same event loop as this promise was created from.
+    ///
+    /// - parameters:
+    ///     - eventLoopBoundValue: The successful result of the operation.
+    @inlinable
+    public func succeed(eventLoopBoundValue: Value) {
+        self._resolve(eventLoopBoundResult: .success(eventLoopBoundValue))
+    }
```

Comment:
I'm not sure this is the right spelling; the pattern we have for functions which must be called on the right EL with runtime assertions are synchronous views (e.g. `ChannelPipeline.SynchronousOperations` and `Channel.syncOptions`). 

It also skirts around the slight oddity with this API in that you could call this function with a `Sendable` i.e. not EL bound value with no ill side effect.

Reply:
I agree. I removed those `loopBound` APIs from this PR since they are really part of the `AssumeIsolated` PR